tbdex/cancel/cancel.go

```diff
@@ -0,0 +1,195 @@
+package cancel
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/TBD54566975/tbdex-go/tbdex/closemsg"
+	"github.com/TBD54566975/tbdex-go/tbdex/crypto"
+	"github.com/TBD54566975/tbdex-go/tbdex/message"
+	"github.com/TBD54566975/tbdex-go/tbdex/validator"
+	"github.com/tbd54566975/web5-go/dids/did"
+	"go.jetpack.io/typeid"
+)
+
+// Kind identifies this message kind
+const Kind = "cancel"
+
+// ValidNext returns the valid message kinds that can follow a cancel.
+func ValidNext() []string {
+	// todo hardcoded orderstatus kind here because otherwise i am introducing a circular dependency :(
```

Comment:
wondering if there's a way around this without hardcoding but could not find think of one myself

Reply:
is cancel not a terminal message? in which case nothing is valid after a cancel yeah?

Reply:
it is not - a cancel after an order indicates the customer wants a refund. if the exchange is subject to refund, then the rest of the exchange looks like PFI sending order statuses to Alice which says `here's the status of your refund` and finally closes the exchange with a Close

or if it's not refundable the PFI can just immediately send a Close which indicates `refund not allowed`

Reply:
ah right, okay thanks!